### DISCUSSION TITLE
Problem with lighting up my LED Strip as an Ambilight with HyperHDR

### DISCUSSION THREAD

**Discussion by Bleyday** [awawa-dev/HyperHDR#101] [Q&A]
Hello dear People,

I am making myself an Ambilight for my TV with a Raspberry Pi 3 and the WS2812B LED Strip with a 5V/15 Amps Power Supply, which should be enough to Power the Pi and LEDs at the same time (as an attempt to fix my Problem, i have also tried to seperate the Power Supply for the LEDs and my Pi). I am doing it with this tutorial here: [https://www.youtube.com/watch?v=J26oYlKyq7Q&t=1519s](https://www.youtube.com/watch?v=J26oYlKyq7Q&t=1519s). I have followed any Step in this Video but the LEDs wont light up. I've checked my Configurations in the HDR Software on the Pi many times the Log seems to be good to me: [HyperHDR Log](https://pastebin.com/g3x639XP) I also attached Screenshots of my HyperHDR Configuration,

Between my TV and the Source is a Capture Card, which is also working properly ( I can see it with the Live Preview of the HyperHDR Software.

At the Pi i am using my GPIO 18 Pin, like as default, to connect to the LED Strip and it is properly connected to the Data Wire of my LEDs, which isnt that long (see Picture). 

Hope someone can help out here, would appreciate
Thank you :)
![capture1](https://user-images.githubusercontent.com/46052824/129066923-e0e65cf9-be3d-4905-a527-8a345cc17df2.jpg)
![capture2](https://user-images.githubusercontent.com/46052824/129066928-f5592c0a-73e3-4382-a973-56718846cff1.jpg)
![conf1](https://user-images.githubusercontent.com/46052824/129066930-04159a50-8f2e-41b5-8520-1acbf7e3194c.jpg)
![conf2](https://user-images.githubusercontent.com/46052824/129066931-e2069ed0-87f1-48d9-8e9a-e1a46f4c27cb.jpg)
![dashboard](https://user-images.githubusercontent.com/46052824/129066935-0a7b2a7e-6f29-4c85-9981-8e63915bf1ec.jpg)
![WhatsApp Image 2021-08-11 at 18 24 37 (1)](https://user-images.githubusercontent.com/46052824/129066952-bf520440-574f-481a-9338-e2298c1cd413.jpeg)
![WhatsApp Image 2021-08-11 at 18 24 37](https://user-images.githubusercontent.com/46052824/129066957-74686aa2-0636-4591-90d4-baf2e88b7ebf.jpeg)

**Comment by awawa-dev** [comment]
Just to note: I know that many tutorials (not ours!) recommends it but using Rpi for driving WS82xx is a bad, bad idea. It's a tempting solution as no other hardware driver is needed but: it's eating resources and then user blames HyperHDR for it ;) , it's fighting with a HyperHDR for a real-time processing which makes things even worse, it's harder to setup and diagnose projects as it can even cause conflicts with some internal Rpi devices or system privileges depending on the RPi model and Linux distribution/version. So recommended solution is a cheap external ESP8266/ESP32 with CH340G/CP2104 and WLED/HyperSerialWLED/HyperSerialESP8266 that can be use over the network or USB cable with Rpi.

Anyway in almost every case it should work with Rpi. Setup&logs are rather OK but:
- after you connected secondary power supply to the RPi then logical levels at the LED strip were probably out of cosmic scale ;) You should try again but this time connect Rpi's pin-out ground to LED strip ground to have a common ground between two power suppliers.
- most users for WS2812B use not SPI but PWM mode...but it requires root privileges on Rpi. And here things starts to be complicated (just warned at the beginning ;) ). You must switch HyperHDR from pi to root user: https://github.com/awawa-dev/HyperHDR/issues/52

**Reply by Bleyday** [reply]
![root](https://user-images.githubusercontent.com/46052824/129080863-07965a06-0f9c-49a1-ba94-3611a1f46c6b.jpg)

i think it is running at root and PWM Mode now and i have reconnected the cables to only one Power supply but its still not running

**Reply by awawa-dev** [reply]
OK, as I said it's a hard thing to diagnose it without ESP. I assume that you check all the connection to the LED strip. Did you measure the voltage on the working system at the LED strip with a multimeter and if there is a connection on the data cable (resistance measure on the powered down system, disconnect other cabled from the LED strip, measure it between input data cable and data point after the first LED on the strip)? 
BTW if it's not a SD image from HyperHDR (not sure, hostname suggested it) then you can start from installing the default image and switch to root again to have a common base and eliminate potential system issues.

**Reply by Bleyday** [reply]
Unfortunately i dont have a multimeter :( and it is already installed on a SD image

**Reply by Bleyday** [reply]
Is it possible to use SPI with Rpi?

**Reply by awawa-dev** [reply]
Just to clarify: I meant full HyperHDR SD image, not a HyperHDR package installed on some other OS. So it seems there are 3-4 option: 
- you can try SD image: zero cost but it's rather unlikely (if you haven't tried it yet)
- buy ESP8266 to be sure in 100% it's a not caused by a software or fried GPIO on the RPI, if it's electrical then in the end you will need a multi-meter
- buy multi-meter to verify connection and voltage (to verify data signal an oscilloscope is needed but let's stops here ;) )

If the LED strip (or even the first LED on the LED strip) is broken then it's a very difficult situation and it's usually a question of faith that could diagnose it: do you have a trust in your multi-meter, ESP board and WLED project to write it off ;) Sometimes additional resistor at the data line helps (WLED suggests 62ohm) but it's rather some kind of last resort solution or magic when everything else fails.

**Reply by awawa-dev** [reply]
> 
> 
> Is it possible to use SPI with Rpi?

you could try other pin meant for PWM on Rpi, maybe output at GPIO18 is gone or there is a system conflict, check your ````dmesg```` anyway.

**Reply by Bleyday** [reply]
> Just to clarify: I meant full HyperHDR SD image, not a HyperHDR package installed on some other OS. So it seems there are 3-4 option:
> 
> * you can try SD image: zero cost but it's rather unlikely (if you haven't tried it yet)
> * buy ESP8266 to be sure in 100% it's a not caused by a software or fried GPIO on the RPI, if it's electrical then in the end you will need a multi-meter
> * buy multi-meter to verify connection and voltage (to verify data signal an oscilloscope is needed but let's stops here ;) )
> 
> If the LED strip (or even the first LED on the LED strip) is broken then it's a very difficult situation and it's usually a question of faith that could diagnose it: do you have a trust in your multi-meter, ESP board and WLED project to write it off ;) Sometimes additional resistor at the data line helps (WLED suggests 62ohm) but it's rather some kind of last resort solution or magic when everything else fails.

Do u have a link where i can get the HyperHDR image?

**Reply by awawa-dev** [reply]
I forgot about level shifter, anyway in most cases it works without it. Guess a matter of luck because it's out of LED specification. If you have an Arduino then it's another way to test it (without level shifter as Arduino already has 5V GPIO)

**Reply by awawa-dev** [reply]
https://github.com/awawa-dev/HyperHDR/releases/tag/v16.0.0.1
SD card images are at the bottom

**Reply by pandel** [reply]
@Bleyday Maybe a silly question, but did you connect your LED strip from the correct side? There should be small arrows on the strip pointing in the direction the data flows. If you connect the data pin from the wrong side nothing will happen. I don't know the WS2812B, but it is the same with the SK6812...

**Reply by awawa-dev** [reply]
@Bleyday As the rpi821x was upgraded recently for the master branch v17 (it has to do with some latest Rpi4 models, should not affect how the old version work when it was properly initialized) needed to some test if it breaks something.  
So to end the confusion for ws821x that configuration is proper when HyperHDR is running as root:  
![obraz](https://user-images.githubusercontent.com/69086569/131249859-5686b7af-50b6-4170-85b7-73c4581d310c.png)
But I've got a voltage level shifter that can make a difference. Or your Rpi's GPIO is fried...
